Fix CheckBox fill prop behavior and type warnings

leossantiago opened the pull request:
<!--- Provide a general summary of the PR in the Title above -->

#### What does this PR do?

Addresses CheckBox `fill` property behavior issues and `type` warnings while using the same prop. 

- `fill` prop behavior

CheckBox component can have the checkbox itself and a label, both wrapped on a div Container. Right now, setting the `fill` prop doesn't result in any changes, where it should expand this Container to the limits of the parent component. 
After some debugging, I ended up discovering that this happens due to fill styling not being properly applied:

https://github.com/grommet/grommet/blob/adfc3091a933958aa5762c9bd493dbe059fb4e61/src/js/components/CheckBox/StyledCheckBox.js#L46-L49

Since `fillStyle` isn't used anywhere else, we can incorporate the `justify-content` prop to it, and call this function if `fill` prop is set.

- `fill` prop type warnings

The warnings mentioned on the issue can easily be associated with the `fill` prop from CheckBox. However, it's related to `StyledCheckBoxContainer`, an internal component that represents the wrapping div of CheckBox. As seen below, it's created based on `styled.label`:

https://github.com/grommet/grommet/blob/adfc3091a933958aa5762c9bd493dbe059fb4e61/src/js/components/CheckBox/StyledCheckBox.js#L41

For some reason, `label` is required to be `true` or `false`, but with the type of string, so, not an issue related to Grommet. Based on that, I added a bypass to that, alongside a Note for future reference so we can update it once it's fixed on `styled-components` end.

I ended up not going further with this as it's not the point of #5172, but those changes made me consider: 
1. Enhancing fill prop to support different alignment options besides `space-between`;
2. Use Box component as base for `StyledCheckBoxContainer` instead of `styled.label`.

Would love to hear our community's inputs on those. 

#### Where should the reviewer start?

`src/js/components/CheckBox/CheckBox.js`
`src/js/components/CheckBox/StyledCheckBox.js`

#### What testing has been done on this PR?

Manual testing and updating existing unit tests. Don't think there's a need to add new ones as the prop usage is already covered.

#### How should this be manually tested?

```js
import React from "react";
import { grommet, Box, CheckBox, Grommet } from "grommet";

export default function App() {
  return (
    <Grommet theme={grommet}>
      <Box fill background="accent-1" pad="small">
        <CheckBox fill reverse label="Check me out" />
      </Box>
    </Grommet>
  );
}
```

#### Any background context you want to provide?

-

#### What are the relevant issues?

#5172 

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?

No.

#### Should this PR be mentioned in the release notes?

#### Is this change backwards compatible or is it a breaking change?

Breaking change, but fixes the property to work as intended.

britt6612 commented:
https://storybook.grommet.io/?path=/story/input-checkbox-inside-a-formfield--inside-form-field 
You can see the change in which the `fill` prop here

britt6612 commented:
@leossantiago I do see the error now: 
<img width="1103" alt="Screen Shot 2021-04-19 at 9 27 58 AM" src="https://user-images.githubusercontent.com/42451602/115262296-a58d5e80-a0f1-11eb-85fc-13eabfc8edcf.png">

britt6612 commented on src/js/components/CheckBox/StyledCheckBox.js:
```diff
@@ -8,6 +8,7 @@ const fillStyle = () => `
       height: 100%;
       max-width: none;
       flex: 1 0 auto;
+      justify-content: space-between;
```
Thank you this is a nice clean up

britt6612 commented on src/js/components/CheckBox/CheckBox.js:
```diff
@@ -195,7 +195,9 @@ const CheckBox = forwardRef(
-        fill={fill}
+        // Note: StyledCheckBoxContainer is based on `styled.label`, which for
+        // some reason requires the `fill` prop to be of type string.
+        fill={fill && 'true'}
```
I think this is good!

ericsoderberghp replied:
If we're trying to avoid a property leaking into the DOM, we should rename it with something like `fillProp`. That allows StyledCheckBoxContainer to have access to the value without react automatically matching it to a DOM attribute.

britt6612 commented on src/js/components/CheckBox/StyledCheckBox.js:
```diff
@@ -43,10 +44,7 @@ const StyledCheckBoxContainer = styled.label`
-    props.fill
-      ? fillStyle() && `justify-content: space-between;`
-      : 'width: fit-content;'}
+  ${props => (props.fill ? fillStyle() : 'width: fit-content;')}
```
This is a nice clean up.. I would maybe add a comment to `fillstyle` that it includes the `justify-content: space-between;` just incase it ends up being used again..

britt6612 commented:
@leossantiago Looks good! Thank you for this! I just dropped one comment!

leossantiago commented:
@britt6612 and @ericsoderberghp just pushed some changes addressing your suggestions! Thank you! 😄 

I'll resolve both comments. And thank you @ericsoderberghp for the heads up on the leaking props. It looks way cleaner now.

britt6612 reviewed (approved):
Better now its not getting leaked to `dom` thank you : 😊

ShimiSun reviewed (approved):
Thank you @leossantiago 💟
